modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -365,27 +373,36 @@ public Response installApplication(@HeaderParam("Accept") String acceptHeader,
 		Message message = new Message();
 
 		try {
-			ConfigOperation configOperation = new ConfigOperation();
-			configOperation.setCode(AndroidConstants.OperationCodes.INSTALL_APPLICATION);
-			configOperation.setType(Operation.Type.CONFIG);
+			InstallApplication installApplication = installApplicationBeanWrapper.getOperation();
 
-			return AndroidAPIUtils.getOperationResponse(applicationBean.getDeviceList(),
-			                                            configOperation, message, responseMediaType);
+			if (installApplication == null) {
+				throw new OperationManagementException("Install application bean is empty");
```

Comment:
Use a better "context-friendly" message rather than saying "Install application *bean* is empty".